Assets/Scripts/ItemComponents/Painter.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Inventory;
+using Mirror;
+using UnityEngine;
+
+namespace ItemComponents
+{
+    /// <summary>
+    /// Component to handle items that can be used to paint with.
+    /// Should be attached to such items.
+    /// </summary>
+    public class Painter : NetworkBehaviour, IItemWithSupply
+    {
+        [SerializeField] private PainterProperties properties = null;
+        [SerializeField] [SyncVar] private float currentSupply;
+
+        public PainterProperties Properties => properties;
+
+        private void Start()
+        {
+            currentSupply = properties.ItemSupply.MaxSupply;
+        }
+        
+        [Server]
+        public void DecreaseSupply()
+        {
+            float newSupply = currentSupply - properties.ItemSupply.DrainRate;
+            if (newSupply < 0)
+            {
+                currentSupply = 0;
+                return;
+            }
+
+            currentSupply = newSupply;
+        }
+
+        [Server]
+        public void IncreaseSupply(float amount)
+        {
+            float newSupply = currentSupply + amount;
+            float maxSupply = currentSupply - properties.ItemSupply.MaxSupply;
+            if (newSupply > maxSupply)
+            {
+                currentSupply = maxSupply;
+                return;
+            }
+
+            currentSupply = newSupply;
```

Comment:
This could all just be replaced by `currentSupply = Mathf.Clamp(currentSupply + amount, 0, properties.ItemSupply.MaxSupply)` in both this and `DecreaseSupply`?

Reply:
You are correct. This is what I get while coding drunk :D